ortb/request_validator.go

```diff
@@ -88,27 +88,32 @@ func (srv *standardRequestValidator) validateImpExt(imp *openrtb_ext.ImpWrapper,
 	prebid := impExt.GetOrCreatePrebid()
 	prebidModified := false
 
+	bidderPromote := false
+
 	if prebid.Bidder == nil {
 		prebid.Bidder = make(map[string]json.RawMessage)
+		bidderPromote = true
 	}
 
 	ext := impExt.GetExt()
 	extModified := false
 
 	// promote imp[].ext.BIDDER to newer imp[].ext.prebid.bidder.BIDDER location, with the later taking precedence
-	for k, v := range ext {
-		if openrtb_ext.IsPotentialBidder(k) {
-			if _, exists := prebid.Bidder[k]; !exists {
-				prebid.Bidder[k] = v
-				prebidModified = true
+	if bidderPromote {
+		for k, v := range ext {
+			if openrtb_ext.IsPotentialBidder(k) {
+				if _, exists := prebid.Bidder[k]; !exists {
+					prebid.Bidder[k] = v
+					prebidModified = true
+				}
+				delete(ext, k)
+				extModified = true
 			}
-			delete(ext, k)
-			extModified = true
 		}
 	}
 
 	if hasStoredAuctionResponses && prebid.StoredAuctionResponse == nil {
-		return []error{fmt.Errorf("request validation failed. The StoredAuctionResponse.ID field must be completely present with, or completely absent from, all impressions in request. No StoredAuctionResponse data found for request.imp[%d].ext.prebid \n", impIndex)}
+		return []error{fmt.Errorf("request validation failed. The StoredAuctionResponse.ID field must be completely present with, or completely absent from, all impressions in request. No StoredAuctionResponse data found for request.imp[%d].ext.prebid", impIndex)}
```

Comment:
While annoying, please keep the trailing new line. This is considered part of the public contract (never know who is relying on a quirk) and reserved for major releases. You're welcomed to open a separate PR for this change and we'll mark it for PBS 4.0.

Checking myself with @bsardo and @bretg for opinions on this.